presto-main/src/main/java/com/facebook/presto/operator/MergeSortProcessor.java

```diff
@@ -144,21 +168,28 @@ public PageWithPosition getPage()
             }
 
             Page page = requireNonNull(pageSupplier.pollPage(), "page is null");
-            currentPage = new PageWithPosition(page);
+            currentPage = new PageWithPosition(streamIndex, page);
             return currentPage;
         }
     }
 
     public static class PageWithPosition
     {
+        private final int streamIndex;
         private final Page page;
         private int position = 0;
 
-        public PageWithPosition(Page page)
+        public PageWithPosition(int streamIndex, Page page)
         {
+            this.streamIndex = streamIndex;
             this.page = requireNonNull(page, "page is null");
         }
 
+        public int getStreamIndex()
```

Comment:
We had similar discussion before. Outside the class encapsulation is ensured by making the inner class to be private. Making this method public has no effect from the compiler perspective,  however it suggest what methods are being used from the outer class, and what methods are used by the inner class only.